### ISSUE TITLE
Calendar layout is not of type JTAppleCalendarMonthLayout.

### ISSUE THREAD

**Issue by EmmaVvv** [patchthecode/JTAppleCalendar#1099]
Updated JTAppleCalendar pod from 7.1.7 to 8.0.0 version. Now I'm getting the following assertion failure: 

`Calendar layout is not of type JTAppleCalendarMonthLayout.
There was an error in this code section. Please contact the developer on GitHub
Assertion failed: file /Users/emma/Documents/appName/Pods/JTAppleCalendar/Sources/JTAppleCalendar/JTACMonthActionFunctions.swift, line 43`

I kinda understand the reason of this issue, most probably the layout has been renamed from `JTAppleCalendarMonthLayout ` to `JTACMonthLayout `, but how can I fix it?

**Comment by patchthecode** [comment]
ok, can you first let me know if you renamed every thing as stated in the guide? (gotta check first)

--> https://patchthecode.com/jtapplecalendar-version-8-0-0-migration-guide/

**Comment by EmmaVvv** [comment]
Yes, I've renamed everything as stated in the migration guide

**Comment by patchthecode** [comment]
can you let me know how you instantiated the calendar?
is it via code? or via Storyboard?

If via code, can you paste it to me?
if via storyboard, can you confirm that your module is correct? 
<img width="465" alt="Screen Shot 2019-07-21 at 2 15 08 pm" src="https://user-images.githubusercontent.com/2439146/61748039-43ad9280-ad6d-11e9-9b47-d3524a1f97b0.png">

**Comment by EmmaVvv** [comment]
I've changed the module to the you mentioned, but now I'm facing many issues inside the pod, which are not covered in Migration guide.

Examples: 

1. Use of unresolved identifier 'selectCounterPartCellIndexPath'
2. Use of unresolved identifier 'addCellToSelectedSet'
3. Use of unresolved identifier 'batchReloadIndexPaths'
etc., etc...

![Screen Shot 2019-07-24 at 1 22 03 AM](https://user-images.githubusercontent.com/53192422/61748525-78433d80-adb1-11e9-974b-91179f743add.png)

**Comment by patchthecode** [comment]
:S 

ok... last resort.
Do a clean (CMD + SHIFT + K)
close Xcode.
reopen build?

**Comment by patchthecode** [comment]
is your project public? If so, email me a copy at patchthecode@gmail.com
if not. I'll have to do some more debugging.

**Comment by EmmaVvv** [comment]
No, the project is private :/ 

And the errors didn't gone :( I also cleaned the derrived data, restarted my Mac and tried to run it again.

I will look forward to hearing from you, have to submit the project to Appstore today :(

**Comment by patchthecode** [comment]
in that case.
I think you should stay on version 7.1.7
The main upgrade to version 8.0.0 is the addition of the YearView mode. And a couple rare bug edge case fixes that no one will find.

I'm heading home from work now. Will be back in about 2hrs.

**Comment by EmmaVvv** [comment]
I will wait for you anyway, I still have about 8 hours, if nothing is possible, will migrate to the previous version

**Comment by patchthecode** [comment]
i hope this was resolved.

**Comment by rupesh4545** [comment]
calendarView.calendarDelegate = self
calendarView.calendarDataSource` = self

put this in your viewdidload :)

**Comment by patchthecode** [comment]
Yea, closing this issue.
Assuming resolved.

**Comment by robwilson195** [comment]
Hi there, also getting this issue. 

I'm currently using v8.0.1 on XCode 10.3.

I instantiated through code, and set the calendar layout as follows: 
` let layout: UICollectionViewFlowLayout = UICollectionViewFlowLayout()
        layout.minimumLineSpacing = 0
        calendar.collectionViewLayout = layout`

I have also tried creating a `JTACMonthLayout` directly to put in here but for some reason XCode keeps giving me `Use of undeclared type 'JTACMonthLayout'`

Look forward to hearing from you, PS sorry if you need more information, I'm new to this.

**Comment by patchthecode** [comment]
@robwilson195 
By any change you can just drop a zip file of your project in this chat? ( assuming your project is just starting here ).

**Comment by robwilson195** [comment]
Yeah sure. Here's a github link too if that's preferable:
https://github.com/robwilson195/tests
[rwTests.zip](https://github.com/patchthecode/JTAppleCalendar/files/3540160/rwTests.zip)

**Comment by robwilson195** [comment]
So I think I might have gotten past the error. I saw in some comment in the pod files that JTAppleCalendarView 'manages it's own layout.' So I removed the UICollectionViewFlowLayout and instead set minimum spacing and cell size with `calendar.cellSize` and `calendar.minimumLineSpacing`. 
I now have a separate issue with the calendar just appearing as a black block, but I'm sure that's just something to do with me not implementing the cells correctly or something. Thanks for taking a look though!
